### PR TITLE
Ruby 3.0.0

### LINKED ISSUE #4777: [ruby] failed to build
Just checking to see if this is really an issue from my installed gcc modifications on x86_64.
Does anyone else have issues rebuilding ruby?

```crew reinstall -s ruby```:

```
compiling ossl_x509crl.c
compiling ../.././ext/psych/yaml/scanner.c
compiling ../.././ext/psych/yaml/writer.c
linking shared-object fiddle.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/fiddle'
compiling ossl_x509ext.c
compiling ossl_x509name.c
compiling ossl_x509req.c
compiling ossl_x509revoked.c
compiling ossl_x509store.c
installing default openssl libraries
linking shared-object psych.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/psych'
linking shared-object openssl.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/openssl'
make[1]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2'
make: *** [uncommon.mk:295: build-ext] Error 2
ruby failed to build: `make V=0 -j4` exited with 2
```

```
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/pty'
make[2]: Entering directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/readline'
compiling readline.c
readline.c: In function 'username_completion_proc_call':
readline.c:79:42: error: 'username_completion_function' undeclared (first use in this function); did you mean 'rl_username_completion_function'?
   79 | # define rl_username_completion_function username_completion_function
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:1904:37: note: in expansion of macro 'rl_username_completion_function'
 1904 |                                     rl_username_completion_function);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:79:42: note: each undeclared identifier is reported only once for each function it appears in
   79 | # define rl_username_completion_function username_completion_function
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:1904:37: note: in expansion of macro 'rl_username_completion_function'
 1904 |                                     rl_username_completion_function);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:250: readline.o] Error 1
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/readline'
make[1]: *** [exts.mk:264: ext/readline/all] Error 2
make[1]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2'
make: *** [uncommon.mk:295: build-ext] Error 2
```

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #4777 
- built with ```-flto```
- key to fixing compile was adding ```-ltinfo```

Works properly:
- [x] x86_64 (for last week or so.)
- [x] armv7l
- [x] i686

### CODE REVIEW

**Comment by satmandu** [comment]
On my system with ruby 2.7.2, ```crew whatprovides bin/ld``` took about 4.7 seconds, and ruby 3.0.0 took about 4.9 seconds.

But the async work in 3.0.0 could help speed things up maybe?

**Comment by uberhacker** [comment]
I very hesitant to merge this since I'm wary of any version with x.0.0. :)  This could break everything in a very bad way.  We also need to update install.sh.

File: packages/ruby.rb [hunk 1]
```diff
@@ -28,12 +28,20 @@ class Ruby < Package
...
+    optflags='-flto' \
+    ./configure #{CREW_OPTIONS} \
+    --enable-shared \
+    --disable-fortify-source"
     system 'make'
   end
 
   def self.check
-    #system "TMPDIR=#{CREW_PREFIX}/tmp make check"
+    system "TMPDIR=#{CREW_PREFIX}/tmp make check || true"
```
Comment (uberhacker) [inline packages/ruby.rb:44]: So does the `|| true` bypass the checks even if they fail?  Test errors are good when building from source, correct?

Comment (satmandu) [reply packages/ruby.rb:44]: Test errors are great, but there are always tests that fail. This still runs the tests, which will outout into compile logs if you make them. (I do.)

But I figure a better test is to just run ruby 3.0.0 for a bit, and I have for several days without any (new) issues I hadn't seen previously. (The ENV issues were showing up for compiles you did as well, and I suspect you aren't using ruby 3.0.)

We aren't really using advanced ruby functionality in any case. (We aren't even installing any gems.)

**Comment by satmandu** [comment]
Also, might make sense to merge this, and NOT update install.sh for a bit. You're not going to break new installs.

**Comment by uberhacker** [comment]
OK, merging but I'm still skeptical. :)

**Pull request merged**